src/effects/effectchainslot.cpp

```diff
@@ -68,6 +68,12 @@ EffectChainSlot::EffectChainSlot(EffectRack* pRack, const QString& group,
 
     connect(&m_channelStatusMapper, SIGNAL(mapped(const QString&)),
             this, SLOT(slotChannelStatusChanged(const QString&)));
+
+    // It is up to skins and controller mappings to do anything useful with this.
+    m_pControlExpanded = new ControlPushButton(ConfigKey(m_group, "expanded"));
```

Comment:
I am not sure if this CO fits quite well here. If it is pure GUI "expanded" control, it can (should) be implemented inside the GUI. 

But it looks like you want to do more with it. It can be uses to switch the effect mappings as well. 

I am not sure if, and how we can achieve this, but I thing it is more a "view_mode" or "knob_layout" enum type. 
The controller might be able to tell Mixxx which mode it supports, or switch between these modes. 
This can be growing by design to be future proof.

What are typical Controller Effect Layouts? 

* Hercules RMX2: Four pressure detecting buttons + one endless knob.
* Denon HS5500: three toggle buttons + one endless knob
* vestax_vci-400: Trakor optimized fx region with mode switch. 
* ....